src/neo/SmartContract/Native/Oracle/OracleContract.cs

```diff
@@ -135,6 +135,13 @@ internal override void Initialize(ApplicationEngine engine)
             engine.Snapshot.Storages.Add(CreateStorageKey(Prefix_RequestId), new StorageItem(BitConverter.GetBytes(0ul)));
         }
 
+        [ContractMethod(0_00100000, CallFlags.None)]
+        private void OnPayment(ApplicationEngine engine, UInt160 from, BigInteger amount)
+        {
+            if (engine.CallingScriptHash != GAS.Hash)
```

Comment:
The calling scripthash couldn't be `GAS.Hash`.

Reply:
Change `CallingScriptHash` after call `CallFromNativeContract`?

Reply:
```csharp
engine.CurrentContext.GetState<ExecutionContextState>().CallingScriptHash = GAS.Hash;
```

Reply:
Or load a GAS context before call it

Reply:
Maybe we add a `MintWithoutCallingOnPayment()` method in GAS.

Reply:
Just remove `PostTransfer` in Mint and Burn method?

Reply:
take a look to https://github.com/Tommo-L/neo/pull/21

Reply:
But it breaks the context, a little....

Reply:
> Just remove PostTransfer in Mint and Burn method?

But you need to fire the `Transfer` event.

Reply:
Please take a look tohttps://github.com/neo-project/neo/pull/2098